dev/UndockedRegFreeWinRT/catalog.cpp

```diff
@@ -0,0 +1,427 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
Is the UndockedRegFreeWinRT code a fork of something else? Or is it brand-new code that needs to be reviewed in this change?

Reply:
It's a fork of [xlang/src/UndockedRegFreeWinRT/UndockedRegFreeWinRT/](https://github.com/microsoft/xlang/tree/master/src/UndockedRegFreeWinRT/UndockedRegFreeWinRT). Changes are:

* Add PackageGraph resolution of WinRT types via appxmanifest.xml
* Renamed URFW functions resolving WinRT types via SxS to BlahBlahBlah_SxS
* Changed WinRTGetThreadingModel and WinRTGetActivationFactory to check PackageGraph before SxS
* Fixed some warnings (URFW not compiled /W4? Different order of #includes? Something slightly differs)
* Changed DllMain logic in dllmain.cpp to a callable API in urfw.*

To focus on the forkage...
1. Review *_PackageGraph() functions and call sites in dev\UndockedRegFreeWinRT\*
2. Compare dev\UndockedRegFreeWinRT\urfw.cpp with xlang's URFW dllmain.cpp
3. Review dev\UndockedRegFreeWinRT\urfw.h

Really should refactor URFW to look at RegFreeWinRT *after* checking the OS, not before, but that requires deep refactoring in multiple places e.g. GetActivationLocation calls WinRTGetThreadingModel() when it really needs to split into PackageGraph then OS then SxS, with parts of GetActivationLocation common, and possibly needing caller to do the whole stack (GetActivationLocation + GetActivationFactory) for PackageGraph then call the OS then whole stack for SxS

So...not today :P One of the things I'm following up with @BenJKuhn and @kythant but only a concern if a type's implemented via RegFreeWinRT *and* the OS (and isn't in the windows.* namespace). So it should be exotic enough not to worry for a while (if ever)